code/game/machinery/iv_drip.dm

```diff
@@ -74,16 +87,22 @@
 		if("eject")
 			eject_beaker()
 			. = TRUE
+		if("detach")
+			if(attached)
+				visible_message(span_notice("[attached] is detached from [src]."))
+				detach_iv()
+			. = TRUE
 		if("changeRate")
-			var/target_rate = params["rate"]
-			if(text2num(target_rate) != null)
-				target_rate = text2num(target_rate)
-				transfer_rate = round(clamp(target_rate, MIN_IV_TRANSFER_RATE, MAX_IV_TRANSFER_RATE), 0.1)
-				. = TRUE
+			if((use_internal_storage || reagent_container) && attached)
+				var/target_rate = params["rate"]
+				if(text2num(target_rate) != null)
+					target_rate = text2num(target_rate)
+					transfer_rate = round(clamp(target_rate, MIN_IV_TRANSFER_RATE, MAX_IV_TRANSFER_RATE), 0.01)
```

Comment:
I really don't like small numbers in reagents, is there any particular reason you're allowing such small quantities to be transferred? 

Also make it a define, it's usually fine but it's reused and actually has a somewhat existent influence on gameplay

Reply:
Thanks, I made it a define.

There are many cases when this small number can be useful:
- to match metabolic rate of some medicines/toxins, they are all fractional
- keep the amount below their overdose thresholds
- to have slow chemical reactions
- to make sort of a timer, when you need to transfer 1 unit per minute